### PR TITLE
Add mechanism to check the code-style

### PR DESCRIPTION
**Pull request by jsaalfeld**
pycodestyle is formerly known as pep8 and one of the widest known style checks for python and jslint is a well-known javascript-linter. With that mechanisms in place it should be easier to maintain a good code quality.

### CODE REVIEW

File: gulpfile.js [hunk 1]
```diff
@@ -66,6 +67,15 @@ gulp.task('js-libs', function () {
         .pipe(gulp.dest(path.join(output_directory, 'js')));
 });
 
+// Lints all js files
```
Comment (normanjaeckel) [inline gulpfile.js]: Please move this task down to our extra tasks.

File: gulpfile.js [hunk 2]
```diff
@@ -26,12 +26,13 @@ var argv = require('yargs').argv,
     templateCache = require('gulp-angular-templatecache'),
     through = require('through2'),
     uglify = require('gulp-uglify'),
-    vsprintf = require('sprintf-js').vsprintf;
+    vsprintf = require('sprintf-js').vsprintf,
+    jshint = require('gulp-jshint'),
```
Comment (normanjaeckel) [inline gulpfile.js]: This line already exists in line 22. Please remove one line.

File: gulpfile.js [hunk 3]
```diff
@@ -26,12 +26,13 @@ var argv = require('yargs').argv,
...
+    stylish = require('jshint-stylish');
```
Comment (normanjaeckel) [inline gulpfile.js]: Please use alphabetic order of imports.

File: gulpfile.js [hunk 4]
```diff
@@ -214,7 +224,8 @@ gulp.task('default', [
         'fonts-libs',
         'ckeditor',
         'angular-chosen-img',
-        'translations'
+        'translations',
+	'js-lint'
```
Comment (normanjaeckel) [inline gulpfile.js]: Please do not run linter on default.

File: gulpfile.js (hunk 1 above)
Comment (normanjaeckel) [reply gulpfile.js]: I see that we already have a js-hint task. So please use (and maybe modify) our task.

File: setup.cfg [hunk 5]
```diff
@@ -27,3 +27,8 @@ disallow_any = unannotated
 
 [mypy-tests.*]
 ignore_errors = true
+
+[pycodestyle]
+max-line-length = 160
+statistics = True
+
```
Comment (normanjaeckel) [inline setup.cfg]: Maybe this line can be removed. Last character in every file should be \n and not \n\n.

**Comment by jsaalfeld** [comment]
We gonna talk about this topic in the next week at a meeting I think

**Pull request closed without merging**